packages/tame-global-error-object/src/main.js

```diff
@@ -1,16 +1,59 @@
-const { getOwnPropertyDescriptor } = Object;
+const { freeze, getOwnPropertyDescriptor, defineProperty } = Object;
+
+/**
+ * hasOwnProperty()
+ * Mimics Object.prototype.hasOwnProperty which uses internally
+ * GetOwnProperty: see https://tc39.es/ecma262/#sec-hasownproperty
+ */
+function hasOwnProperty(obj, prop) {
+  return getOwnPropertyDescriptor(obj, prop) !== undefined;
+}
+
+function assert(condition, message) {
+  if (!condition) {
+    throw new TypeError(message);
+  }
+}
 
 export default function tameGlobalErrorObject() {
-  // Tame static properties.
-  delete Error.captureStackTrace;
+  // Tame static methods
+
+  if (hasOwnProperty(Error, 'captureStackTrace')) {
+    const { captureStackTrace } = {
+      captureStackTrace(targetObject, _constructorOpt) {
```

Comment:
These two lines seem weird.. why not just say `function captureStackTrace`? Is this a clever trick to avoid `this` somehow?

Reply:
It's to create a concise method that has no [[construct]] behavior and no `prototype` property.

Reply:
```
// Before patching
console.log(Error.captureStackTrace.name); // captureStackTrace
console.log(Error.captureStackTrace.prototype); // undefined

// Patching with a function
Error.captureStackTrace = function captureStackTrace() {}
console.log(Error.captureStackTrace.name); // 'captureStackTrace' 
console.log(Error.captureStackTrace.prototype); // {} => NOT OK!

// Patching with an arrow function
Error.captureStackTrace = () => {}
console.log(Error.captureStackTrace.name); // '' => NOT OK! 
console.log(Error.captureStackTrace.prototype); // undefined

// Patching with a concise method
const ErrorStatic = { captureStackTrace() {} };
Error.captureStackTrace = ErrorStatic.captureStackTrace;
console.log(Error.captureStackTrace.name); // 'captureStackTrace'
console.log(Error.captureStackTrace.prototype); // undefined
```

Reply:
Very nice @erights!

For documentation, this is the toString()
```js
// Before patching
console.log(Error.captureStackTrace.toString());
// Output:
// function captureStackTrace() { [native code] }

// Patching with a function
Error.captureStackTrace = function captureStackTrace() {}
console.log(Error.captureStackTrace.toString());
// Output:
// function captureStackTrace() {}

// Patching with an arrow function
const captureStackTrace = () => {}
Error.captureStackTrace = captureStackTrace;
console.log(Error.captureStackTrace.toString());
// Output:
// () => {}

// Patching with a concise method
const ErrorStatic = { captureStackTrace() {} };
Error.captureStackTrace = ErrorStatic.captureStackTrace;
console.log(Error.captureStackTrace.toString());
// Output:
// captureStackTrace() {}
```

As long as we prevent regressions by clearly commenting and guarding this side effect with a test, I think the arrow function is better. 

It's clear that the output of the native `toString()` is not executable code,  and we should not expect to mimic the specs automagically. Therefore, I'm fine with the output of `toString()` with either form, it's not supposed to be evaluable anyway if we follow specs. And if we want, we need custom code anyways.